### PR TITLE
have ct_eval(Identifier) return a handle if appropiate

### PR DESCRIPTION
**Pull request by SebastianMestre**
@lushoBarlett 

This patch makes what you are working on a bit simpler, I think. Hopefully you can make use of it!

### CODE REVIEW

File: src/typecheck.cpp [hunk 1]
```diff
@@ -87,22 +87,13 @@ void typecheck(TypedAST::Identifier* ast, TypeChecker& tc) {
...
-	} else if (meta_type == tc.meta_typefunc()) {
-		assert(0 && "Accessed a name of a typefunc (not implemented)");
-		// we are a type function.
-		// TODO: not too sure what needs to be done...
-	} else {
-		assert(0 && "Accessed a name with unknown metatype (not implemented)");
-		// meta type var
-		// TODO: not too sure what needs to be done...
-	}
+	assert(meta_type == tc.meta_value());
```
Comment (lushoBarlett) [inline src/typecheck.cpp:91]: Okaaaaay, now this makes a lot more sense. I guess I was confused as to what the purpose and responsibility differences were between metacheck, ct_eval and typecheck.

File: src/ct_eval.cpp [hunk 2]
```diff
@@ -33,6 +33,45 @@ TypedAST::FunctionLiteral* ct_eval(
...
+	assert(ast);
+	assert(ast->m_declaration);
+
+	// TODO: These two lines are fishy; I copied them from
+	// elsewhere, but I don't think we need to go through
+	// the declaration to get the metatype: this is already
+	// done for us by metacheck().
+	MetaTypeId meta_type =
+	    tc.m_core.m_meta_core.find(ast->m_declaration->m_meta_type);
+	ast->m_meta_type = meta_type;
```
Comment (lushoBarlett) [inline src/ct_eval.cpp:53]: All of this is already done in metacheck, yes. But not quite. In metacheck there is no find. The identifier just points to the declaration's meta type... I guess this lines can still be deleted?

**Review by lushoBarlett — APPROVED** [review]
Looks like it will do the trick. Now in typecheck, all info on types, I can get from Handles.

File: src/ct_eval.cpp (hunk 2 above)
Comment (SebastianMestre) [reply src/ct_eval.cpp:53]: That makes sense to me. I will merge it as-is, and look into it later today

**Pull request merged**